store/secretsmgr.go

```diff
@@ -0,0 +1,246 @@
+
+//SecretsManagerClientIface defines the parts of the client we are actually using, for testing purposes
+type SecretsManagerClientIface interface {
+	secretsmanager.ListSecretsAPIClient
+	GetSecretValue(ctx context.Context, params *secretsmanager.GetSecretValueInput, optFunc ...func(options *secretsmanager.Options)) (*secretsmanager.GetSecretValueOutput, error)
+	CreateSecret(ctx context.Context, params *secretsmanager.CreateSecretInput, optFunc ...func(*secretsmanager.Options)) (*secretsmanager.CreateSecretOutput, error)
+	PutSecretValue(ctx context.Context, params *secretsmanager.PutSecretValueInput, optFunc ...func(*secretsmanager.Options)) (*secretsmanager.PutSecretValueOutput, error)
+	DeleteSecret(ctx context.Context, params *secretsmanager.DeleteSecretInput, optFunc ...func(*secretsmanager.Options)) (*secretsmanager.DeleteSecretOutput, error)
+}
+
+type SecretsManager struct {
+	logger              log.Logger
+	client              SecretsManagerClientIface
+	SecretRetentionDays int64 //time in days to keep a deleted secret
+	DefaultTimeout      time.Duration
+}
+
+func NewSecretsManager(client *secretsmanager.Client, secretsRetentionDays int64, logger log.Logger, defaultTimeout time.Duration) (Store, error) {
+	if secretsRetentionDays != 0 && (secretsRetentionDays < 7 || secretsRetentionDays > 30) {
+		return nil, errors.New("aws only supports post-deletion retention periods of between 7 and 30 days. Use 0 to force it to delete immediately")
+	}
+
+	return &SecretsManager{
+		logger:              logger,
+		client:              client,
+		SecretRetentionDays: secretsRetentionDays,
+		DefaultTimeout:      defaultTimeout,
+	}, nil
+}
+
+func (s *SecretsManager) timeoutContext() (context.Context, context.CancelFunc) {
+	return context.WithTimeout(context.Background(), s.DefaultTimeout)
+}
+
+/*
+Get returns the most recent value of the given secret. NOTE that only string type secrets are supported
+at the moment, not binary type.
+*/
+func (s *SecretsManager) Get(service Service, name string) (Parameter, error) {
+	ctx, cancelFunc := s.timeoutContext()
+	defer cancelFunc()
+
+	secretNameWithPath := fmt.Sprintf("%s/%s", service.Prefix(), name)
+	req := &secretsmanager.GetSecretValueInput{
+		SecretId: &secretNameWithPath,
+	}
+
+	response, err := s.client.GetSecretValue(ctx, req)
+	if err != nil {
+		return Parameter{}, err
+	}
+
+	safeValue := ""
+	if response.SecretString != nil {
+		safeValue = *response.SecretString
+	}
+
+	return Parameter{
+		Name:     *response.Name,
+		Value:    safeValue,
+		IsSecret: true,
+		Service:  service,
+	}, nil
+}
+
+/*
+fetchSecretValue returns the most recent version of the listed SecretListEntry.
+This two-stage retrieval is necessary because the List call does _not_ decrypt the
+secrets.
+NOTE: we assume that secrets are strings, not binary values, for the time being
+*/
+func (s *SecretsManager) fetchSecretValue(entry *types.SecretListEntry) (string, error) {
+	ctx, cancelFunc := s.timeoutContext()
```

Comment:
I think timeout handling is probably worth introducing in a separate PR. And add the same behaviour to SSM too at the same time. Probably the best way to do this is by adding an initial `ctx` arg to all of the `Store` interface operations, as this is consistent with e.g. the AWS libraries and elsewhere.